Source/WebCore/dom/ScriptElement.cpp

```diff
@@ -370,6 +394,37 @@ bool ScriptElement::requestModuleScript(const TextPosition& scriptStartPosition)
     return true;
 }
 
+bool ScriptElement::requestImportMap(Frame& frame, const String& sourceURL)
+{
+    ASSERT(m_element.isConnected());
+    ASSERT(!m_loadableScript);
+    if (!stripLeadingAndTrailingHTMLSpaces(sourceURL).isEmpty()) {
+        auto script = LoadableImportMap::create(m_element.nonce(), m_element.attributeWithoutSynchronization(HTMLNames::integrityAttr), referrerPolicy(),
+            m_element.attributeWithoutSynchronization(HTMLNames::crossoriginAttr), m_element.localName(), m_element.isInUserAgentShadowTree(), hasAsyncAttribute());
+
+        auto scriptURL = m_element.document().completeURL(sourceURL);
+        m_element.document().willLoadScriptElement(scriptURL);
+
+        const auto& contentSecurityPolicy = *m_element.document().contentSecurityPolicy();
+        if (!contentSecurityPolicy.allowNonParserInsertedScripts(scriptURL, URL(), m_startLineNumber, m_element.nonce(), String(), m_parserInserted))
+            return false;
+
+        frame.script().setPendingImportMaps();
+        if (script->load(m_element.document(), scriptURL)) {
+            m_loadableScript = WTFMove(script);
+            m_isExternalScript = true;
+        }
+    }
+
+    if (m_loadableScript)
+        return true;
+
+    callOnMainThread([this, element = Ref<Element>(m_element)] {
+        dispatchErrorEvent();
```

Comment:
The behavior itself is valid since it is aligned to the spec https://html.spec.whatwg.org/multipage/scripting.html#prepare-a-script
> If src is the empty string, then [queue a task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-task) to [fire an event](https://dom.spec.whatwg.org/#concept-event-fire) named [error](https://html.spec.whatwg.org/multipage/indices.html#event-error) at el, and return.

ScriptElement is inherited by multiple script element classes, and `dispatchErrorEvent` is a virtual function implemented by them, so we need to call that (otherwise the overridden function will be ignored). Default implementation of `dispatchErrorEvent` is dispatching an event to m_element. (SVGScriptElement has additional logic).

> Also usually, when dispatching an event, we queue a task on the document. Can we do that here?

This is nice! Changed it to use document's EventLoop's queueTask

Reply:
If we need 'this', should we capture a Ref then?